instrumentation/struts-2.3/src/test/java/io/opentelemetry/javaagent/instrumentation/hypertrace/struts/StrutsInstrumentationTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The Hypertrace Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.javaagent.instrumentation.hypertrace.struts;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.opentelemetry.sdk.trace.data.SpanData;
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import javax.servlet.DispatcherType;
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.apache.struts2.dispatcher.ng.filter.StrutsPrepareAndExecuteFilter;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.DefaultServlet;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.util.resource.FileResource;
+import org.hypertrace.agent.core.instrumentation.HypertraceSemanticAttributes;
+import org.hypertrace.agent.testing.AbstractInstrumenterTest;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class StrutsInstrumentationTest extends AbstractInstrumenterTest {
+
+  private static final String REQUEST_BODY = "hello";
+  private static final String REQUEST_HEADER = "requestheader";
+  private static final String REQUEST_HEADER_VALUE = "requestvalue";
+  private static final String RESPONSE_HEADER = "headerName";
+  private static final String RESPONSE_HEADER_VALUE = "headerValue";
+
+  private static Server server = new Server(0);
+  private static int serverPort;
+
+  @BeforeAll
+  public static void startServer() throws Exception {
+    ServletContextHandler handler = new ServletContextHandler();
+    FileResource resource = new FileResource(StrutsInstrumentationTest.class.getResource("/"));
+    handler.setContextPath("/context");
+    handler.setBaseResource(resource);
+    handler.addServlet(DefaultServlet.class, "/");
+    handler.addFilter(
+        StrutsPrepareAndExecuteFilter.class, "/*", EnumSet.allOf(DispatcherType.class));
+    server.setHandler(handler);
+    server.start();
+    serverPort = server.getConnectors()[0].getLocalPort();
+  }
+
+  @AfterAll
+  public static void stopServer() throws Exception {
+    server.stop();
+  }
+
+  @Test
+  public void testBody() throws IOException, TimeoutException, InterruptedException {
```

Comment:
we don't use `test` prefix in the test methods. The name for this could be `postUrlEncoded`